### PR TITLE
Add gas estimate calculations and safe processing of `out of gas` error

### PR DESCRIPTION
**Pull request by antonina-cherednichenko**

### CODE REVIEW

File: poster/src/post_with_retries.ts [hunk 1]
```diff
@@ -55,6 +59,14 @@ async function postWithRetries(transaction: TransactionConfig, signerKey: string
       };
     }
 
+    if (maybeIsOutOfGas(e)) {
```
Comment (antonina-cherednichenko) [inline poster/src/post_with_retries.ts:62]: I added a gas increase ability for the situation when a gas estimate is not very accurate

**Comment by jflatow** [comment]
Looks great

File: poster/src/post_with_retries.ts [hunk 2]
```diff
@@ -55,6 +59,14 @@ async function postWithRetries(transaction: TransactionConfig, signerKey: string
...
+      console.log('Transaction is possibly out of gas, increasing gas limit');
+      transaction = {
+        ...transaction,
+        gas: Math.floor(Number(transaction.gas) * GAS_ADJUSTMENT)
```
Comment (jflatow) [inline poster/src/post_with_retries.ts:66]: Maybe we should always apply at least one level of adjustment?

Comment (antonina-cherednichenko) [reply poster/src/post_with_retries.ts:66]: @coburncoburn @hayesgm any thoughts here? I can increase the gas limit on each retry to simplify logic here

**Pull request merged**